poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd

```diff
@@ -0,0 +1,166 @@
+author: "Denisse Fierro Arcos"
+date: "2023-06-07"
+output: 
+  github_document:
+    toc: true
+    html_preview: false
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Goal of this notebook
+
+This notebook will demonstrate how to use one or more Great Barrier Reef (GBR) features recognised by the Great Barrier Reef Marine Park Authority (GBRMPA) dataset to extract water temperature data from the AIMS Sea Surface Temperature Monitoring Program available at RIMReP `geoparquet` collection.
+
+# Loading libraries
+
+```{r libraries, results = "hide", message = F, warning = F}
+library(arrow)
+library(dplyr)
+library(magrittr)
+library(stringr)
+library(ggplot2)
+library(sf)
+library(lubridate)
+source("useful_spatial_functions_data_extraction.R")
+```
+
+# Connecting to RIMReP collection
+
+Connecting to the AIMS Sea Surface Temperature Monitoring Program. This can take a minute or so.
+
+```{r connect}
+#Establishing connection
+data_bucket <- s3_bucket("s3://rimrep-data-public/091-aims-sst/test-50-64-spatialpart")
+
+#Accessing dataset
+data_df <- open_dataset(data_bucket)
+```
+
+# Extracting monitoring sites and their coordinates
+
+```{r sites_bin}
+sites_shp <- data_df %>% 
+  #We select unique sites included in the dataset
+  distinct(site, subsite, lon, lat) %>%
+  #This will load them into memory
+  collect()
+
+#Creating shapefile of unique locations
+sites_shp <- sites_shp %>% 
+  #Creating column to identify deployment location
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA)) %>%
+  #Turning into shapefile
+  st_as_sf(coords = c("lon", "lat"), crs = 4326)
+
+#Checking results
+glimpse(sites_shp)
+```
+# Identifying AIMS sites within GBR feature of interest
+In this section, we will use the functions included in the [useful_spatial_functions_data_extraction](https://github.com/aodn/rimrep-examples/blob/main/poc-data-api/useful_spatial_functions_data_extraction.R) script to load GBR features of interest. We will then use the boundaries of the feature of interest to identify the names of the sites within these limits.
+
+We will use the `gbr_features` function to access the boundaries for our sites of interest. If no parameters are given to this function, it will load all features recognised by the GBRMPA. You could also provide the name of the site using the `site_name` parameter or the GBRMPA unique ID using the `site_id` parameter. You can also use both parameters at the same time in case you have the names for some sites and the IDs for others.
+
+As an example, we will use **U/N Reef** or unnamed reefs. We chose them simply because they cover a large area and will overlap with the most number of AIMS monitoring sites.

```

Comment:
In fact there are multiple reefs without a name, and they are referred to as **U/N Reef**, but each of them has a `UNIQUE_ID` value. The use of **U/N Reef** as a filter will return multiple reefs in the GBR